From 1683cddcb21bb1839c99e3c4a4a1361b4c25a622 Mon Sep 17 00:00:00 2001
From: Gu Zitao <guzitao@wxiat.com>
Date: Tue, 22 Feb 2022 22:12:41 +0800
Subject: [PATCH] sw64: add sw_64 support to buildtar, mkdebian and builddeb
Patch-mainline: NO, OTHERS
References: OLK-5.10
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4SPZD
CVE: NA

-------------------------------

In order to support build deb package on sw64 architecture, add sw_64
case.

Signed-off-by: Gu Zitao <guzitao@wxiat.com> #openEuler_contributor
Signed-off-by: Laibin Qiu <qiulaibin@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 scripts/package/builddeb | 3 +++
 scripts/package/buildtar | 3 +++
 scripts/package/mkdebian | 2 +-
 3 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index 91a502bb97e8..cd558ad20128 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -126,6 +126,9 @@ um)
 parisc|mips|powerpc)
 	installed_image_path="boot/vmlinux-$version"
 	;;
+sw_64)
+	installed_image_path="boot/vmlinux.bin-$version"
+	;;
 *)
 	installed_image_path="boot/vmlinuz-$version"
 esac
diff --git a/scripts/package/buildtar b/scripts/package/buildtar
index 936198a90477..a0e4c7be89b6 100755
--- a/scripts/package/buildtar
+++ b/scripts/package/buildtar
@@ -92,6 +92,9 @@ case "${ARCH}" in
 	alpha)
 		[ -f "${objtree}/arch/alpha/boot/vmlinux.gz" ] && cp -v -- "${objtree}/arch/alpha/boot/vmlinux.gz" "${tmpdir}/boot/vmlinuz-${KERNELRELEASE}"
 		;;
+	sw_64)
+		[ -f "${objtree}/arch/sw_64/boot/vmlinux.bin" ] && cp -v -- "${objtree}/arch/sw_64/boot/vmlinux.bin" "${tmpdir}/boot/vmlinux-bin-${KERNELRELEASE}"
+		;;
 	parisc*)
 		[ -f "${KBUILD_IMAGE}" ] && cp -v -- "${KBUILD_IMAGE}" "${tmpdir}/boot/vmlinux-${KERNELRELEASE}"
 		[ -f "${objtree}/lifimage" ] && cp -v -- "${objtree}/lifimage" "${tmpdir}/boot/lifimage-${KERNELRELEASE}"
diff --git a/scripts/package/mkdebian b/scripts/package/mkdebian
index 60a2a63a5e90..fdef3f585dc0 100755
--- a/scripts/package/mkdebian
+++ b/scripts/package/mkdebian
@@ -26,7 +26,7 @@ set_debarch() {
 
 	# Attempt to find the correct Debian architecture
 	case "$UTS_MACHINE" in
-	i386|ia64|alpha|m68k|riscv*)
+	i386|ia64|alpha|m68k|riscv*|sw_64)
 		debarch="$UTS_MACHINE" ;;
 	x86_64)
 		debarch=amd64 ;;
-- 
2.26.2

